File: Cloude/Cloude/Utilities/Theme.swift
// Theme.swift

import SwiftUI

enum DS {
    enum Text {
        static let m: CGFloat = 13
        static let s: CGFloat = 10
    }

    enum Icon {
        static let s: CGFloat = 14
        static let m: CGFloat = 16
        static let l: CGFloat = 19
    }

    enum Spacing {
        static let xs: CGFloat = 4
        static let s: CGFloat = 8
        static let m: CGFloat = 12
        static let l: CGFloat = 16
        static let xl: CGFloat = 24
        static let xxl: CGFloat = 32
    }

    enum Radius {
        static let s: CGFloat = 6
        static let m: CGFloat = 9
        static let l: CGFloat = 12
    }

    enum Size {
        static let m: CGFloat = 28
        static let l: CGFloat = 44
        static let xxl: CGFloat = 200
    }

    enum Scale {
        static let s: CGFloat = 0.6
        static let m: CGFloat = 0.9
        static let l: CGFloat = 1.2
    }

    enum Stroke {
        static let s: CGFloat = 0.6
        static let m: CGFloat = 1.2
        static let l: CGFloat = 1.8
    }

    enum Duration {
        static let s: Double = 0.2
        static let m: Double = 0.5
        static let l: Double = 0.8
    }

    enum Opacity {
        static let s: Double = 0.15
        static let m: Double = 0.4
        static let l: Double = 0.7
    }
}

private struct AppThemeKey: EnvironmentKey {
    static let defaultValue: AppTheme = .majorelle
}

extension EnvironmentValues {
    var appTheme: AppTheme {
        get { self[AppThemeKey.self] }
        set { self[AppThemeKey.self] = newValue }
    }
}

struct ThemePalette {
    let background: UInt
    let secondary: UInt
}

enum AppTheme: String, CaseIterable {
    case monet = "Monet"
    case turner = "Turner"
    case malevich = "Malevich"
    case bauder = "Bauder"
    case majorelle = "Majorelle"
    case klimt = "Klimt"

    static var current: AppTheme {
        if let raw = UserDefaults.standard.string(forKey: "appTheme"),
           let theme = AppTheme(rawValue: raw) {
            return theme
        }
        return .majorelle
    }

    var colorScheme: ColorScheme {
        switch self {
        case .monet, .turner: return .light
        default: return .dark
        }
    }
}

File: android/app/src/main/java/com/cloude/app/Utilities/Theme.kt
package com.cloude.app.Utilities

import androidx.compose.ui.unit.dp
import androidx.compose.ui.unit.sp

object DS {
    object Text {
        val s = 10.sp
        val m = 13.sp
    }

    object Icon {
        val s = 14.dp
        val m = 16.dp
        val l = 19.dp
    }

    object Spacing {
        val xs = 4.dp
        val s = 8.dp
        val m = 12.dp
        val l = 16.dp
        val xl = 24.dp
        val xxl = 32.dp
    }

    object Radius {
        val s = 6.dp
        val m = 9.dp
        val l = 12.dp
    }

    object Size {
        val m = 28.dp
        val l = 44.dp
        val xxl = 200.dp
    }

    object Scale {
        const val s = 0.6f
        const val m = 0.9f
        const val l = 1.2f
    }

    object Stroke {
        val s = 0.6.dp
        val m = 1.2.dp
        val l = 1.8.dp
    }

    object Duration {
        const val s = 200
        const val m = 500
        const val l = 800
    }

    object Opacity {
        const val s = 0.15f
        const val m = 0.4f
        const val l = 0.7f
    }
}

data class ThemePalette(
    val background: Long,
    val secondary: Long
)

enum class AppTheme(val displayName: String) {
    Monet("Monet"),
    Turner("Turner"),
    Malevich("Malevich"),
    Bauder("Bauder"),
    Majorelle("Majorelle"),
    Klimt("Klimt");

    val isLight: Boolean
        get() = this == Monet || this == Turner

    val palette: ThemePalette
        get() = when (this) {
            Monet -> ThemePalette(background = 0xFFFFFFFF, secondary = 0xFFF2F2F7)
            Turner -> ThemePalette(background = 0xFFFDF6E3, secondary = 0xFFEEE8D5)
            Malevich -> ThemePalette(background = 0xFF000000, secondary = 0xFF0A0A0A)
            Bauder -> ThemePalette(background = 0xFF131A24, secondary = 0xFF1A2332)
            Majorelle -> ThemePalette(background = 0xFF0C0F1F, secondary = 0xFF141A35)
            Klimt -> ThemePalette(background = 0xFF141008, secondary = 0xFF221A0C)
        }
}
